search/search.go

```diff
@@ -183,12 +183,12 @@ type DocumentMatch struct {
 	ScoreBreakdown map[int]float64 `json:"score_breakdown,omitempty"`
 
 	// internal variable used in PreSearch phase of search in alias
-	// indicated the index id of the index that this match came from
-	// used in vector search.
-	// it is a stack of index ids, the top of the stack is the index id
+	// to indicate the name of the index that this match came from.
+	// used in knn search.
+	// it is a stack of index names, the top of the stack is the name
 	// of the index that this match came from
 	// of the current alias view, used in alias of aliases scenario
-	IndexId []int `json:"index_id,omitempty"`
+	IndexNames []string `json:"index_names,omitempty"`
```

Comment:
I worry about the noise this is going to generate to the end user.

Reply:
this stack will not be output to the end user, since when the doc match reaches the index in phase 2, this slice will always be empty, it is similar to score breakdown, both score breakdown and indexNames will be used between the 2 phases only, the final result will be sans them

Reply:
That's good!